typing/rec_check.ml

```diff
@@ -818,6 +855,35 @@ let rec expression : Typedtree.expression -> term_judg =
     | Texp_open (od, e) ->
       open_declaration od >> expression e
 
+(* Function bodies.
+
+    G |-{body} b : m
+*)
+and function_body body =
+  match body with
+  | Tfunction_body body ->
+    (*
+        G |- e : m
+        ------------------
+        G |-{body} e : m (**)
+
+      (**) The "e" here stands for [Tfunction_body] as opposed to
+           [Tfunction_cases].
```

Comment:
(I think that this precision is sort of redundant with the placement of the comment inside the `Tfunction_body` case, but do as you prefer, I am not one to complain about long comments.)